sentry node docs (#119)
oasislabs/docs: src/operators/sentry-node.md

# level, you can use:
#
# log:
#   level: debug
log:
  level:
    # Per-module log levels. Longest prefix match will be taken. Fallback to
    # "default", if no match.
    default: debug
    tendermint: warn
    tendermint/context: error
  format: JSON
  # By default logs are output to stdout. If you're running this in docker keep the
  # default
  #file: /var/log/oasis-node.log

# Path to the genesis file for the current version of the network.
genesis:
  file: /serverdir/etc/genesis.json

worker:
  sentry:
    # Enable sentry node.
    enabled: true
    # Port used by validator nodes to query sentry node for registry information.
    # IMPORTANT: Only validator nodes protected by the sentry node should have
    # access to this port. This port should not be exposed on the public network.
    control_port: 9009

# Tendermint backend configuration.
tendermint:
  core:
    listen_address: tcp://0.0.0.0:26656
    external_address: tcp://{{ external_address }}:26656

  seeds: "{{ seed_node_address }}"

  # Sentry node should set validator IDs as private peer IDs.
  private_peer_id:
    - "{{ validator_tendermint_id }}"
```

variables present in the above configuration file:

> `{{ validator_tendermint_id }}`

Reviewer: That's correct. See https://github.com/oasislabs/docs/blob/master/src/operators/joining-the-testnet.md for examples.

> `{{ validator_tendermint_id }}`

Author: yeah this is not at the state yet where i would bother with how it will render yet :-)

> `{{ validator_tendermint_id }}`

Reviewer 2: This issue has been a paint point for some time: vuejs/vuepress#853.

I've used `<code v-pre>{{ vars }}</code>` because it is possible to include such text inline.
And I've just omitted the `{{` and `}}` parts when explaining the variables.

# Sentry Node Architecture

This guide provides instructions for a deployment using the Sentry node
architecture to protect validator nodes from being directly exposed on the
public network.

This guide assumes a setup where an Oasis validator node is only accessible
over a private network, with sentry nodes having access to it.
The guide does not cover setting this infrastructure up.
Knowledge of [Tendermint's Sentry Node architecture][tm-sentry] is assumed as
well.

[tm-sentry]: https://cosmos.network/docs/cosmos-hub/validators/security.html#sentry-nodes-ddos-protection

::: danger

> This is an example of a Sentry node deployment and we take no responsibility

Reviewer: ```suggestion
This is only an example of a Sentry node deployment, and we take no responsibility
```

for mistakes contained therein.
Make sure you understand what you are doing.
:::

## Prerequisites

Before following this guide, make sure you've read the [Prerequisites] and
[Joining the Testnet] guides and created your Entity.

[Prerequisites]: ./prerequisites.md
[Joining the Testnet]: ./joining-the-testnet.md

## Configuring the Oasis Sentry Node

### Initializing Sentry Node

Sentry node identity keys can be initialized with:

```
$ oasis-node registry node init --entity /localhostdir/entity
```

The generated `tls_identity_cert.pem` (which is the node's TLS cert for
securing gRPC connections) needs to be available on any node that will be
running behind the sentry node.

::: tip NOTE
We plan to simplify this step in the future so that it won't involve Entity at
all since Sentry node's identity keys are in no way linked to the Entity
itself.
See [Oasis Core #2413](https://github.com/oasislabs/oasis-core/issues/2413) for
more details.
:::

### Configuring Sentry Node

> Oasis node can be configured to run as a sentry node by setting the

Reviewer: ```suggestion
An Oasis node can be configured to run as a sentry node by setting the
```

    # "default", if no match.
    default: debug
    tendermint: warn
    tendermint/context: error
  format: JSON
  # By default logs are output to stdout. If you're running this in docker keep
  # the default
  #file: /var/log/oasis-node.log

# Path to the genesis file for the current version of the network.
genesis:
  file: /serverdir/etc/genesis.json

# Worker configuration.
worker:
  sentry:
    # Enable sentry node.
    enabled: true
    # Port used by validator nodes to query sentry node for registry
    # information.
    # IMPORTANT: Only validator nodes protected by the sentry node should have
    # access to this port. This port should not be exposed on the public
    # network.
    control_port: 9009

# Tendermint backend configuration.
tendermint:
  core:
    listen_address: tcp://0.0.0.0:26656
    external_address: tcp://{{ external_address }}:26656

  seeds: "{{ seed_node_address }}"

  # Sentry node should set validator IDs as private peer IDs.
  private_peer_id:
    - "{{ validator_tendermint_id }}"
```

::: tip NOTE
Multiple sentry nodes can be provisioned following the above steps.
:::

## Configuring the Oasis Validator Node

> In this setup the Oasis validator should not be exposed directly on the public

Reviewer: ```suggestion
In this setup the Oasis validator node should not be exposed directly on the public
```

> network. The Oasis validator only needs access to its sentry nodes, preferably

Reviewer: ```suggestion
network. The Oasis validator only needs to be able to connect to its sentry nodes, preferably
```
